kubernetes/structures_pod.go

```diff
@@ -661,6 +661,18 @@ func flattenReadinessGates(in []v1.PodReadinessGate) ([]interface{}, error) {
 
 // Expanders
 
+func expandPodTargetState(p []interface{}) []string {
+	if len(p) > 0 {
+		t := make([]string, len(p))
+		for i, v := range p {
+			t[i] = v.(string)
+		}
+		return t
+	}
+
+	return []string{string(v1.PodRunning)}
```

Comment:
Shouldn't this rather be done with a `Default:` or `DefaultFunc:` attribute in the schema? 
Feels like setting the default here makes it less obvious that there is a default value.

Reply:
This is actually what I have stuck with and ended up with the current solution:

- I can't do `Default: []string{string(corev1.PodRunning)}`. In this case, `apply` ends up with the error message saying that `Default` is not valid for lists or sets.
- When I do `Default: string(corev1.PodRunning),` on the `Elem` level, it doesn't work too. This is what `Default` documentation says about this **_Default cannot be used if the Schema is directly an implementation of an Elem field of another Schema, such as trying to set a default value for a TypeList or TypeSet._**
- `DefaultFunc` doesn't work too. I tried `DefaultFunc: func() (interface{}, error) {return []string{string(corev1.PodRunning)}, nil}` on the List level and `DefaultFunc: func() (interface{}, error) {return string(corev1.PodRunning), nil}` on the `Elem` level. In both cases, `d.Get("target_state")` returns an empty slice.

Maybe I miss something.

Reply:
I don't think you're missing anything. The List and Set types in the SDK have always had problems with defaults.

Let's just keep it the way you did it then, it also covers the case where `target_state` is set to an empty string or nil.